core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -72,7 +72,8 @@ private[spark] class ReliableCheckpointRDD[T: ClassTag](
       .sortBy(_.toString)
     // Fail fast if input files are invalid
     inputFiles.zipWithIndex.foreach { case (path, i) =>
-      if (!path.toString.endsWith(ReliableCheckpointRDD.checkpointFileName(i))) {
+      if (path.getName != ReliableCheckpointRDD.checkpointFileName(i) &
```

Comment:
Why can't you accept both types of file name this way every where they're looked for / read? that seems best of all. Maybe I am missing something.